### DISCUSSION TITLE
Rust implementation co2mon

### DISCUSSION THREAD

**Discussion by vit1251** [dmage/co2mon#53] [Show and tell]
I want to express my gratitude for the incredible code you've shared. It has proven to be immensely helpful! Based on the provided code, an alternative library has also been developed in Rust (see https://github.com/vit1251/rs-co2mon), which could be useful for users looking for a safe and performant option.

Additionally, I would like to inquire about the documentation you used to decode the protocol. There are still some questions regarding the message packets numbered 67, 79, 86, 87, 109, 110, and 113. It would be fantastic if we could decode those as well.

Thank you once again for your valuable contributions!

**Comment by dmage** [comment]
It could be random noise just to make reverse engineering more difficult.

**Reply by dmage** [reply]
https://revspace.nl/CO2MeterHacking - this one is mentioned in README, check it out if you haven't seen it yet.
